app/controllers/customers_controller.rb

```diff
@@ -0,0 +1,8 @@
+class CustomersController < ApplicationController
+    def show
+        @customer = Customer.find(params[:id])
+        @item_id = params[:item_id]
```

Comment:
Instead of routing the form back to the show page, it's more RESTful to route it route it to a controller action that better describes what the functionality is doing -- *creating* a joins record between an existing customer and item. 

In the pair project, you did end up routing some forms back to this show action, but that was for search functionality, in order to retrieve some data and render it to the user so they could decide whether to adopt the pet or not. You were not making a change to the database in that case (just reading from it). Or, in relational rails, there are stories where you can click a button and see the same data ordered alphabetically. That could also be routed back to the show action because you're again just affecting what will be rendering on the page to the user. Again, that's just a read action: you're querying the data and just displaying it differently. 

In the case of this form, you're adding an item to the customer, which does actually lead to the database being updated rather than just querying. In that case, that's when we want the form to route to a different action, in this case a `create` action, and then it can redirected back to the show page. 

So - in summary - here is a rule of thumb. If my form is:
- querying data and formatting it (maybe in a particular way, like filtering, ordering, etc) -> form can route to show action
- making new records, updating records, etc but then maybe redirecting back to same page -> form routes to a different controller and a specific controller action for that purpose, like `create` or `update`